scripts/coccinelle/reserved_function_names.cocci

```diff
@@ -0,0 +1,98 @@
+/// Find function names starting with '__' or '_' and rename them
+// Copyright: (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
// Confidence: High ?

Just check the reports carefully as this script
changes the codebase massivefully.

Reply:
Agree

Reply:
Seems to work better now, need to still check the output of this script before setting any confidence level.

Reply:
Let's fix the confidence level after another round of updates. There are quite a few _impl_* functions that are somehow missed.